CMSgov/easi-app
# [EASI-3479] Soft delete support for TRB Recommendations and Documents

# EASI-3479

## Changes and Description

- Modifies the logic for deleting TRB Advice Letter Recommendations and TRB Documents to be soft deletes instead of hard deletes

This logic helps support the functionality in https://github.com/CMSgov/easi-app/pull/2254, in that users will be able to delete TRB Recommendations / Documents if they have a note attached to them, while still maintaining their FKEY.

## How to test this change

- `scripts/dev db:seed`
- Documents
  - Try uploading & deleting documents (and checking the DB to see that they're still there)
  - Ensure no views of documents render any deleted documents (except notes that reference them)
- Recommendations
  - Try deleting, adding, and re-ordering recommendations to ensure that order is preserved properly when soft deleting
  - Ensure no views of recommendations show deleted entities (except notes that reference them)

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
